### ISSUE TITLE
[DataObject] LocalizedField nested in Block causes error when containing data

### ISSUE THREAD

**Issue by rberneder** [pimcore/pimcore#9221]
In the class definitions it is possible to nest a ```LocalizedField``` in a ```Block```. Objects of that class can be opened and saved without any problems. But as soon as any data field in the ```LocalizedField``` contains any data, the object panel of the Pimcore backend UI breaks.
Also the folder preview and the folder tree preview no longer work when such a "corrupt" shall listed.

Pimcore X v10.0.0

The stack trace:
Fatal error:  Uncaught TypeError: Cannot assign Pimcore\Model\Element\ElementDescriptor to property Pimcore\Model\DataObject\Localizedfield::$object of type ?Pimcore\Model\DataObject\Concrete in /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php:252
Stack trace:
#0 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): ReflectionProperty->setValue(Object(Pimcore\Model\DataObject\Localizedfield), Object(Pimcore\Model\Element\ElementDescriptor))
#1 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(Pimcore\Model\DataObject\Localizedfield), Object(ReflectionProperty))
#2 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(Pimcore\Model\DataObject\Localizedfield))
#3 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Localizedfield))
#4 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(Pimcore\Model\DataObject\Data\BlockElement), Object(ReflectionProperty))
#5 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(Pimcore\Model\DataObject\Data\BlockElement))
#6 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#7 /var/www/html/vendor/pimcore/pimcore/models/DataObject/Data/BlockElement.php(241): DeepCopy\DeepCopy->copy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#8 /var/www/html/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(594): Pimcore\Model\DataObject\Data\BlockElement->marshalForCache()
#9 [internal function]: Pimcore\Cache\Core\CoreCacheHandler->Pimcore\Cache\Core\{closure}(Object(Pimcore\Model\DataObject\Data\BlockElement))
#10 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/TypeFilter/ReplaceFilter.php(28): call_user_func(Object(Closure), Object(Pimcore\Model\DataObject\Data\BlockElement))
#11 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(125): DeepCopy\TypeFilter\ReplaceFilter->apply(Object(Pimcore\Model\DataObject\Data\BlockElement))
#12 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy->recursiveCopy(Object(Pimcore\Model\DataObject\Data\BlockElement))
#13 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy->copyArray(Array)
#14 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(155): DeepCopy\DeepCopy->recursiveCopy(Array)
#15 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(135): DeepCopy\DeepCopy->copyArray(Array)
#16 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(252): DeepCopy\DeepCopy->recursiveCopy(Array)
#17 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(208): DeepCopy\DeepCopy->copyObjectProperty(Object(App\Model\Product\Car), Object(ReflectionProperty))
#18 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(144): DeepCopy\DeepCopy->copyObject(Object(App\Model\Product\Car))
#19 /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php(94): DeepCopy\DeepCopy->recursiveCopy(Object(App\Model\Product\Car))
#20 /var/www/html/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(605): DeepCopy\DeepCopy->copy(Object(App\Model\Product\Car))
#21 /var/www/html/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(910): Pimcore\Cache\Core\CoreCacheHandler->storeCacheData('object_81', Object(App\Model\Product\Car), Array, NULL, false)
#22 /var/www/html/vendor/pimcore/pimcore/lib/Cache/Core/CoreCacheHandler.php(958): Pimcore\Cache\Core\CoreCacheHandler->writeSaveQueue()
#23 /var/www/html/vendor/pimcore/pimcore/lib/Cache.php(200): Pimcore\Cache\Core\CoreCacheHandler->shutdown(false)
#24 /var/www/html/vendor/pimcore/pimcore/lib/Pimcore.php(240): Pimcore\Cache::shutdown()
#25 /var/www/html/vendor/pimcore/pimcore/lib/Kernel.php(242): Pimcore::shutdown()
#26 [internal function]: Pimcore\Kernel->Pimcore\{closure}()
#27 {main}
thrown in /var/www/html/vendor/myclabs/deep-copy/src/DeepCopy/DeepCopy.php on line 252

**Comment by weisswurstkanone** [comment]
@rberneder could you please provide your class export ?

**Comment by rberneder** [comment]
Sure, here you go @weisswurstkanone. I have tested this class on the current Pimcore demo site and observed the same error.

```json
{
    "id": "test_class",
    "description": "",
    "modificationDate": 1621962365,
    "parentClass": "",
    "implementsInterfaces": "",
    "listingParentClass": "",
    "useTraits": "",
    "listingUseTraits": "",
    "allowInherit": false,
    "allowVariants": false,
    "showVariants": false,
    "layoutDefinitions": {
        "fieldtype": "panel",
        "layout": null,
        "border": false,
        "name": "pimcore_root",
        "type": null,
        "region": null,
        "title": null,
        "width": 0,
        "height": 0,
        "collapsible": false,
        "collapsed": false,
        "bodyStyle": null,
        "datatype": "layout",
        "permissions": null,
        "childs": [
            {
                "fieldtype": "panel",
                "layout": null,
                "border": false,
                "name": "Layout",
                "type": null,
                "region": null,
                "title": "",
                "width": "",
                "height": "",
                "collapsible": false,
                "collapsed": false,
                "bodyStyle": "",
                "datatype": "layout",
                "permissions": null,
                "childs": [
                    {
                        "fieldtype": "block",
                        "lazyLoading": false,
                        "disallowAddRemove": false,
                        "disallowReorder": false,
                        "collapsible": false,
                        "collapsed": false,
                        "maxItems": null,
                        "styleElement": "",
                        "childs": [
                            {
                                "fieldtype": "localizedfields",
                                "childs": [
                                    {
                                        "fieldtype": "input",
                                        "width": "",
                                        "defaultValue": null,
                                        "columnLength": 190,
                                        "regex": "",
                                        "unique": false,
                                        "showCharCount": false,
                                        "name": "title",
                                        "title": "title",
                                        "tooltip": "",
                                        "mandatory": false,
                                        "noteditable": false,
                                        "index": false,
                                        "locked": false,
                                        "style": "",
                                        "permissions": null,
                                        "datatype": "data",
                                        "relationType": false,
                                        "invisible": false,
                                        "visibleGridView": false,
                                        "visibleSearch": false
                                    }
                                ],
                                "name": "localizedfields",
                                "region": null,
                                "layout": null,
                                "title": "",
                                "width": "",
                                "height": "",
                                "maxTabs": null,
                                "border": false,
                                "provideSplitView": false,
                                "tabPosition": null,
                                "hideLabelsWhenTabsReached": null,
                                "fieldDefinitionsCache": null,
                                "permissionView": null,
                                "permissionEdit": null,
                                "tooltip": "",
                                "mandatory": false,
                                "noteditable": false,
                                "index": null,
                                "locked": false,
                                "style": "",
                                "permissions": null,
                                "datatype": "data",
                                "relationType": false,
                                "invisible": false,
                                "visibleGridView": true,
                                "visibleSearch": true,
                                "labelWidth": "",
                                "labelAlign": "left"
                            }
                        ],
                        "layout": null,
                        "fieldDefinitionsCache": null,
                        "name": "gallery",
                        "title": "gallery",
                        "tooltip": "",
                        "mandatory": false,
                        "noteditable": false,
                        "index": false,
                        "locked": false,
                        "style": "",
                        "permissions": null,
                        "datatype": "data",
                        "relationType": false,
                        "invisible": false,
                        "visibleGridView": false,
                        "visibleSearch": false
                    }
                ],
                "locked": false,
                "icon": "",
                "labelWidth": "",
                "labelAlign": "left"
            }
        ],
        "locked": false,
        "icon": null,
        "labelWidth": 100,
        "labelAlign": "left"
    },
    "icon": "",
    "previewUrl": "",
    "group": "",
    "showAppLoggerTab": false,
    "linkGeneratorReference": "",
    "previewGeneratorReference": "",
    "compositeIndices": [],
    "generateTypeDeclarations": true,
    "showFieldLookup": false,
    "propertyVisibility": {
        "grid": {
            "id": true,
            "key": false,
            "path": true,
            "published": true,
            "modificationDate": true,
            "creationDate": true
        },
        "search": {
            "id": true,
            "key": false,
            "path": true,
            "published": true,
            "modificationDate": true,
            "creationDate": true
        }
    },
    "enableGridLocking": false
}
```

**Comment by weisswurstkanone** [comment]
regression of #8684

**Comment by dvesh3** [comment]
Fixed by #9228
